appstore/developer_portal_api.py

```diff
@@ -0,0 +1,629 @@
+        if requestOK[0] == True:
+
+            params = dict(request.form)
+
+            screenshots = {
+                "aplite": [],
+                "basalt": [],
+                "chalk": [],
+                "diorite": []
+            }
+
+            try:
+                pbw_file = request.files['pbw'].read()
+                pbw = PBW(pbw_file, 'aplite')
+            except Exception as e:
+                return jsonify(error = f"Your pbw file is corrupt or invalid", e = "invalid.pbw"), 400
+
+            try:
+                with pbw.zip.open('appinfo.json') as f:
+                    appinfo = json.load(f)
+            except Exception as e:
+                return jsonify(error = f"Your pbw file is invalid or corrupt", e = "invalid.pbw"), 400
+
+
+            appinfo_valid = is_valid_appinfo(appinfo)
+            if not appinfo_valid[0]:
+                return jsonify(error = f"The appinfo.json in your pbw file has the following error: {appinfo_valid[1]}", e = "invalid.appinfocontent"), 400
+            
+            # Check app doesn't already exist
+            try:
+                if App.query.filter(App.app_uuid == appinfo['uuid']).count() > 0:
+                    return jsonify(error = "An app already exists with that UUID", e = "app.exists"), 400
+            except Exception as e:
+                return jsonify(error = "The UUID provided in appinfo.json is invalid", e = "invalid.uuid"), 400
+
+            # Get developer ID from auth (This is also where we check the user is authenticated)
+            result = authed_request('GET', f"{config['REBBLE_AUTH_URL']}/api/v1/me/pebble/appstore")
+            if result.status_code != 200:
+                abort(401)
+            me = result.json()
+            developer_id = me['id']
+
+            # Find developer
+            developer = Developer.query.filter(Developer.id == developer_id).one_or_none()
+
+            if developer is None:
+                return jsonify(error = "You do not have an active developer account.", e = "account.invalid", message = "Please visit dev-portal.rebble.io to activate your developer account"), 409
+
+            # Upload banner if present
+            if "banner" in request.files:
+                header_asset = upload_asset(request.files["banner"], request.files["banner"].content_type)
+            else:
+                header_asset = None
+
+            # Copy screenshots to platform map
+            for platform in screenshots:
+                for x in range(6):
+                    if f"screenshot-{platform}-{x}" in request.files:
+                        screenshots[platform].append(request.files[f"screenshot-{platform}-{x}"])
+
+            # Remove any platforms with no screenshots
+            clearedScreenshots = dict(screenshots)
+            for platform in screenshots:
+                if len(clearedScreenshots[platform]) < 1:
+                    del clearedScreenshots[platform]
+            screenshots = clearedScreenshots
```

Comment:
That said, this is cleaner:

```suggestion
            screenshots = {k: v for k, v in screenshots.items() if v}
```